### ISSUE TITLE
AutoModel produces different results with SentenceTransformer and FlagEmbedding

### ISSUE THREAD

**Issue by yuzhichang** [FlagOpen/FlagEmbedding#1312]
Run following script:
```
#!/usr/bin/env python3
import torch
from transformers import AutoModel, AutoTokenizer
from sentence_transformers import SentenceTransformer
from FlagEmbedding import BGEM3FlagModel

sentence = "What is Deep Learning?"
# sentence = "Hello world!"

def local_inference_flagembedding():
    '''https://github.com/FlagOpen/FlagEmbedding/blob/master/examples/inference/embedder/README.md#3-m3-model'''
    model = BGEM3FlagModel(
        'BAAI/bge-m3',
        devices="cpu",
        pooling_method='cls',
        cache_dir="huggingface.co",
    )
    
    embeddings = model.encode(
        [sentence],
        return_dense=True,
        return_sparse=False,
        return_colbert_vecs=False,
        convert_to_numpy=True,
    )
    print("Local inference via FlagEmbedding:",  embeddings["dense_vecs"])

def local_inference_huggingface():
    '''https://github.com/FlagOpen/FlagEmbedding/blob/master/examples/inference/embedder/README.md#2-m3-model'''
    model_name = "BAAI/bge-m3"
    model = AutoModel.from_pretrained(model_name)
    tokenizer = AutoTokenizer.from_pretrained(model_name)
    model.eval()

    # 准备输入数据
    inputs = tokenizer([sentence], padding=True, truncation=True, return_tensors="pt")

    # 验证输出
    with torch.no_grad():
        output = model(**inputs)
        print("Local inference via AutoModel:", output[0][:, 0].numpy())

def local_inference_huggingface2():
    '''https://github.com/FlagOpen/FlagEmbedding/blob/master/examples/inference/embedder/README.md#using-sentence-transformers'''
    encoder = SentenceTransformer("BAAI/bge-m3")
    embeddings = encoder.encode(sentence, normalize_embeddings=True)
    print("Local inference via SentenceTransformer:", embeddings)


if __name__ == "__main__":
    local_inference_flagembedding()
    local_inference_huggingface()
    local_inference_huggingface2()
    print("PASS")
```

The output is:
```
Local inference via FlagEmbedding: [[-0.04269356 -0.05327216 -0.01346236 ... -0.03036356  0.03309653 -0.00372823]]
Local inference via AutoModel: [[-1.1328763  -1.4135802  -0.35722458 ... -0.80569893  0.8782184 -0.09892879]]
Local inference via SentenceTransformer: [-0.04269356 -0.05327216 -0.01346236 ... -0.03036356  0.03309653 -0.00372823]
```

SentenceTransformer and FlagEmbedding produce the same result. However AutoModel produces different ones.

**Comment by ZiyiXia** [comment]
I think that's because the AutoModel way is not normalizing the final embedding vector. 
`BGEM3FlagModel` has `normalize_embeddings=True` by default and the `SentenceTransformer` part set `normalize_embeddings=True` in encode().

**Comment by yuzhichang** [comment]
With following normalization, the result matchs with BGEM3FlagModel and SentenceTransformer.
```
    with torch.no_grad():
        output = model(**inputs)
        embeddings = output[0][:, 0]
        embeddings = torch.nn.functional.normalize(embeddings, dim=-1)
        print("Local inference via AutoModel:", embeddings.numpy())

```
